### ISSUE TITLE
Does the order matter for importing tensorflow and keras?

### ISSUE THREAD

**Issue by ZhaoWang-IIT** [tensorflow/tensorflow#28572]
**System information**
- OS Platform and Distribution: Linux Ubuntu 16.04;
- I create a conda virtual environment:
conda create -n tf_gpu tensorflow-gpu
conda activate tf_gpu
- TensorFlow installed in conda virtual environment: 
conda install -c conda-forge/label/cf201901 
- TensorFlow version: 1.14.1-dev20190424;
- Python version: 3.6.7;

**Problem (1)**
I am running python in linux terminal and get errors if I change the order of importing keras and tensorflow. Is there any explanation for why the following happens? What is the dependency relation between tensorflow and keras?

No error if I import keras first and then tensorflow:
![image](https://user-images.githubusercontent.com/13983910/57488299-50dfd880-7278-11e9-8b60-78ac24a38f63.png)

Got errors if I import tensorflow first and then keras (error occurs for both importing tensorflow and keras):
![image](https://user-images.githubusercontent.com/13983910/57488524-f2672a00-7278-11e9-9ca9-b69d54778af3.png)

![image](https://user-images.githubusercontent.com/13983910/57488548-03b03680-7279-11e9-96fd-c71d4200fe7c.png)

**Problem (2)**
- I add the virtual environment to my jupyter kernel by:
```
source activate tf_gpu
python -m ipykernel install --user --name tf_gpu
```
- I create a jupyter notebook and run the commands:
```
import keras 
import tensorflow
```
The order of importing keras and tensorflow doesn't matter in jupyter notebook.

**My question is:** 
Why does the order of importing keras and tensorflow matter when I am running python script, but the order doesn't matter when I am running a jupyter notebook?

**Comment by achandraa** [comment]
No need to import Keras and TensorFlow independently. When we import tensorflow, keras will also get imported automatically no matter whether you use jupyter notebook or terminal. Let us know if that resolves the issue.Thanks!

**Comment by mihaimaruseac** [comment]
Note that `keras` can be used without a proper tensorflow install. In your first screenshot that's exactly what happens: `import keras` gives a warning that it is using the Theano backend. Then you `import tensorflow` and override the previous `keras` import.

**Comment by achandraa** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
